plugins/sinks/shield/sink.go

```diff
@@ -0,0 +1,198 @@
+package shield
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/odpf/meteor/models"
+	assetsv1beta2 "github.com/odpf/meteor/models/odpf/assets/v1beta2"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/registry"
+	"github.com/odpf/salt/log"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+//go:embed README.md
+var summary string
+
+type Config struct {
+	Host    string            `mapstructure:"host" validate:"required"`
+	Headers map[string]string `mapstructure:"headers"`
+}
+
+var info = plugins.Info{
+	Description: "Send user information to shield http service",
+	Summary:     summary,
+	Tags:        []string{"http", "sink"},
+	SampleConfig: heredoc.Doc(`
+	# The hostname of the shield service
+	host: https://shield.com
+	# Additional HTTP headers send to shield, multiple headers value are separated by a comma
+	headers:
+	  X-Shield-Email: meteor@odpf.io
+      X-Other-Header: value1, value2
+	`),
+}
+
+type httpClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type Sink struct {
+	plugins.BasePlugin
+	client httpClient
```

Comment:
We can use GRPC to connect to shield. Here is an example in optimus extractor.
https://github.com/odpf/meteor/blob/main/plugins/extractors/optimus/optimus.go

Reply:
Changed to GRPC-based client.